src/flamenco/runtime/fd_executor.c

```diff
@@ -746,7 +746,12 @@ fd_execute_instr( fd_exec_txn_ctx_t * txn_ctx,
     fd_exec_txn_ctx_reset_return_data( txn_ctx );
     int exec_result = FD_EXECUTOR_INSTR_SUCCESS;
     if( native_prog_fn != NULL ) {
-      exec_result = native_prog_fn( *ctx );
+      fd_borrowed_account_t * prog_acct = &txn_ctx->borrowed_accounts[instr->program_id];
+      if( memcmp( prog_acct->const_meta->info.owner, fd_solana_native_loader_id.uc, sizeof(fd_pubkey_t) ) ) {
+        exec_result = FD_EXECUTOR_INSTR_ERR_UNSUPPORTED_PROGRAM_ID;
```

Comment:
Sorry to be pedantic, but maybe we should add a comment here where this check and error code comes from (since it's a dubious loaded program cache thing)

Reply:
This code comes from no place.  It was invented on the spot and 100% not required since it is basically waisting time checking an invariant that, if violated, would be a disaster..